ReportCardParser/src/reportcardparser/ReportCardStats.java

```diff
@@ -43,6 +63,40 @@ public static String GetBestSubject(ReportCard reportCard) {
      */
     public static double SubjectAverage(ReportCard[] reportCards, String subject) {
         // REPLACE ME WITH A REAL IMPLEMENTATION!
-        return 0.0;
+       int subjectIndex = 0;
```

Comment:
What if reportCards is null (or the array entries are null)?  What if the subject is not found for a particular student?